fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java

```diff
@@ -77,6 +78,10 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
         predicates = rangeExtractor.rewriteOnlyColumn(Utils.compoundAnd(Utils.extractConjuncts(predicates)
                 .stream().map(rangeExtractor::rewriteOnlyColumn).collect(Collectors.toList())));
         Preconditions.checkState(predicates != null);
+
+        // simplify case-when
+        predicates = scalarOperatorRewriter.rewrite(predicates, Lists.newArrayList(new SimplifiedCaseWhenRule()));
+        predicates = Utils.pruneTediousPredicate(predicates);
```

Comment:
put `SimplifiedCaseWhenRule` in `ScalarOperatorRewriter.DEFAULT_REWRITE_SCAN_PREDICATE_RULES`